states.go

```diff
@@ -348,8 +349,8 @@ func stateError(sr *StateRequest) error {
 // A utility function to fetch the appropriate state of the vehicle
 func (c *Client) fetchState(resource string, id int64) (*StateRequest, error) {
 	stateRequest := &StateRequest{}
-	if err := c.getJSON(
-		c.baseURL+"/vehicles/"+strconv.FormatInt(id, 10)+"/data_request"+resource, stateRequest); err != nil {
+	path := strings.Join([]string{c.baseURL, "vehicles", strconv.FormatInt(id, 10), "data_request", resource}, "/")
```

Comment:
`path.Join` ?

Reply:
I was using path.Join, but it was doing something weird in tests. It seems to strip the double / from the protocol? Didn't seem "right" so I used strings.Join.

https://play.golang.org/p/EEF_iUtUIGK